src/nimlsp.nim

```diff
@@ -387,11 +387,13 @@ while true:
                 # Try to guess the size of the identifier
                 let
                   message = diagnostic.nimDocstring
-                  endcolumn = diagnostic.column +  message.rfind('\'') - message.find('\'') - 1
+                  endcolumn = max(diagnostic.column + message.rfind('\'') - message.find('\'') - 1, 0)
+                  diagnosticLine = max(diagnostic.line-1, 0)
+                  diagnosticColumn = max(diagnostic.column, endcolumn)
                 response.add create(Diagnostic,
                   create(Range,
-                    create(Position, diagnostic.line-1, diagnostic.column),
-                    create(Position, diagnostic.line-1, max(diagnostic.column, endcolumn))
+                    create(Position, diagnosticLine, diagnosticColumn),
```

Comment:
This should always be `diagnostic.column`, and not `endColumn`. This logic was stolen from the VSCode plugin which tries to find the thing that was wrong and underline it. So the range here is supposed to go from where Nim finds the issue, and to the end of the thing that is in error.